Assets/Scripts/Checkpoint.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Checkpoint : MonoBehaviour
+{
+    [SerializeField]
+    private float unactivatedRotationSpeed = 100, activatedRotationSpeed=300;
+
+    [SerializeField]
+    private float unactivatedScale = 1, activatedScale = 1.5f;
+
+    [SerializeField]
+    private Color unactivatedColor, activatedColor;
+
+    private bool isActivated = false;
+    private SpriteRenderer spriteRenderer;
+
+    private void Start()
+    {
+        spriteRenderer = GetComponent<SpriteRenderer>();
+        UpdateColor();
+    }
+
+    private void Update()
+    {
+        UpdateRotation();
+    }
+
+    private void UpdateColor()
+    {
+        Color color = unactivatedColor;
+        if (isActivated)
+        {
+            color = activatedColor;
+        }
+        spriteRenderer.color=color;
```

Comment:
put a space between the words and the equal sign, just for style/readability reasons.